mucommander-core/src/main/java/com/mucommander/search/SearchDialog.java

```diff
@@ -248,6 +270,32 @@ private void cleanError() {
         showDialog();
     }
 
+
+    void addSizePanel(YBoxPanel mainPanel) {
+        YBoxPanel sizePanel = new YBoxPanel(10);
+        sizePanel.setBorder(BorderFactory.createTitledBorder(Translator.get("Size")));
+        XAlignedComponentPanel compPanel = new XAlignedComponentPanel(5);
+        GridBagConstraints gbc = ProportionalGridPanel.getDefaultGridBagConstraints();
+        gbc.weightx = 1.0;
+        JPanel sz = new ProportionalGridPanel(4, gbc);
+        sz.add(new JLabel(""));
+
+        cboSizeRel.setSelectedIndex(lastSizeRel.index);
+        sz.add(cboSizeRel);
+
+        tfSize.setText(lastSize == null ? "" : lastSize.toString());
+        sz.add(tfSize);
+
+        cboSizeUnit.setSelectedIndex(lastSizeUnit.index);
+        sz.add(cboSizeUnit);
+
+        compPanel.addRow("", sz, 10);
+        sizePanel.add(compPanel);
+        mainPanel.add(sizePanel);
+        mainPanel.addSpace(10);
+    }
```

Comment:
yeah, this layout is not that nice :)
how about adding it as another row in the File Search panel with label `Size` and the component will be a panel with BorderLayout - the relation will be in the west, the size at the center and the unit at the east?

Reply:
I've played a little bit with the layouts (BorderLayout looks ugly,  GridLayout looks less ugly), now I have found one solution which is not as ugly as the other ones.